src/main/java/uk/gov/digital/ho/hocs/info/api/TeamService.java

```diff
@@ -206,6 +207,24 @@ public void addUsersToTeam(List<UUID> userUUIDs, UUID teamUUID) {
         log.info("Added users with UUIDs {} to team with UUID {}", userUuidList, team.getUuid().toString(), value(EVENT, USERS_ADDED_TO_TEAM));
     }
 
+    @Transactional
+    public void patchTeam(UUID teamUUID, PatchTeamDto patchTeamDto) {
+        log.debug("Patching team {}", teamUUID);
+        Team team = getTeam(teamUUID);
+
+        if (!Objects.isNull(patchTeamDto.getDisplayName())
+                && !patchTeamDto.getDisplayName().equals(team.getDisplayName())) {
+            this.updateTeamName(teamUUID, patchTeamDto.getDisplayName());
+        }
+
+        if (!Objects.isNull(patchTeamDto.getUnitUUID())) {
+            UUID newUnitUuid = UUID.fromString(patchTeamDto.getUnitUUID());
+            if(!team.getUnit().getUuid().equals(newUnitUuid)) {
+                moveToNewUnit(newUnitUuid, teamUUID);
+            }
+        }
+    }
+
```

Comment:
why is one if using && and the other two ifs?

Reply:
We originally needed to check if a UUID string was present, then convert it into a UUID object before comparing it. I've changed this now so the DTO uses UUID instead of String (Jackson handles this), so it's no longer necessary.